lxpanel-plugin/kano_profile.c

```diff
@@ -96,6 +125,20 @@ static GtkWidget *plugin_constructor(LXPanel *panel, config_setting_t *settings)
 
     gtk_widget_set_sensitive(icon, TRUE);
 
+    /* Start watching the pipe for input. */
+
+    // Profile file
+    gchar *filename = get_profile_filename();
+    if (filename) {
+        plugin->status_file = g_file_new_for_path(filename);
```

Comment:
I may be missing something but it looks like you `g_free` the actual status file but not the `filename` variable.